src/type_parser.h

```diff
@@ -12,43 +12,47 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#ifndef SRC_FORMAT_PARSER_H_
-#define SRC_FORMAT_PARSER_H_
+#ifndef SRC_TYPE_PARSER_H_
+#define SRC_TYPE_PARSER_H_
 
 #include <memory>
 #include <string>
 #include <vector>
 
-#include "src/format.h"
+#include "src/type.h"
 
 namespace amber {
 
-/// Parses a Vulkan image string into a format object.
-class FormatParser {
+/// Parses a Vulkan image string into a type object.
+class TypeParser {
  public:
-  static FormatType NameToType(const std::string& data);
+  static FormatType NameToFormatType(const std::string& data);
 
-  FormatParser();
-  ~FormatParser();
+  TypeParser();
+  ~TypeParser();
 
-  std::unique_ptr<Format> Parse(const std::string& fmt);
+  std::unique_ptr<type::Type> Parse(const std::string& fmt);
```

Comment:
This could probably use a comment to explain what format the string is expected to have.

Reply:
done.